src/Inciter/CharmMesh.cpp

```diff
@@ -26,33 +26,29 @@
   #pragma clang diagnostic pop
 #endif
 
-using tk::CharmMesh;
+using inciter::CharmMesh;
 
-CharmMesh::CharmMesh(
-  const std::vector< std::size_t >& inpoel,
-  const tk::UnsMesh::Coords& coord,
-  const std::map< int, std::vector< std::size_t > >& bface,
-  const std::vector< std::size_t >& triinpoel,
-  const std::map< int, std::vector< std::size_t > >& bnode ) :
-  m_inpoel( inpoel ),
-  m_coord( coord ),
-  m_bface( bface ),
-  m_triinpoel( triinpoel ),
-  m_bnode( bnode )
+CharmMesh::CharmMesh( const std::vector< std::size_t >& inpoel,
+                      const tk::UnsMesh::Coords& coord,
+                      const tk::Fields& u,
+                      int nchare ) :
+  m_inpoel( inpoel ), m_coord( coord ), m_u( u )
 // *****************************************************************************
 //  Constructor
 //! \param[in] inpoel Vector of mesh element connectivity owned (local IDs)
 //! \param[in] coord Coordinates of mesh nodes
-//! \param[in] bface Boundary face lists mapped to side set ids
-//! \param[in] triinpoel Triangle element connectivity
-//! \param[in] bnode Boundary node lists mapped to side set ids
+//! \param[in] u Solution vector to transfer
+//! \param[in] nchare Number of chares in chare array
 // *****************************************************************************
 {
   Assert( !inpoel.empty(), "No elements assigned to CharmMesh chare" );
   Assert( tk::positiveJacobians( m_inpoel, m_coord ),
           "Jacobian in input mesh to CharmMesh non-positive" );
   Assert( tk::conforming( m_inpoel, m_coord ),
           "Input mesh to CharmMesh not conforming" );
+
+  exam2m::addMesh( thisProxy, nchare,
```

Comment:
This function should just be called once per mesh. By having it in the constructor it is being called once per mesh-chare. You could put it inside `if (thisIndex == 0)` if you still want to call it here. You'd then also want the CkCallback to be a broadcast to `thisProxy` rather than just a p2p message to `thisProxy[thisIndex]`